### DISCUSSION TITLE
In the context of `mini.pairs`, what do `<CR>` and `<BS>` represent or perform?

### DISCUSSION THREAD

**Discussion by jiggol** [nvim-mini/mini.nvim#1372] [Q&A] [answered]
Hello, I am struggling to understand the meaning of the phrase "Pairs get automatically registered to be recognized by <BS> and <CR>" in the description of mini.pairs.

I live in a non-English-speaking country. Could you please provide additional explanation about what kind of actions this refers to?

Thank you very much for creating such a great plugin.

**Comment by echasnovski** [comment, marked as answer]
No worries, the documentation probably could be a bit more explicit here.

`<CR>` means `Enter` key and `<BS>` means `Backspace` key. It is a standard notation for those keys in Neovim help. You can look at `:h key-notation` for more details.

Registering pair for the key means that pressing that key when inside the pair will do something extra (see [here](https://github.com/echasnovski/mini.nvim/blob/c8922aef8207137e66c80acdb9523668599ba62a/doc/mini-pairs.txt#L165-L168)):
- Pressing `<CR>` inside registered pair (for example, `()`) leads to putting an extra blank indented line between pair characters. See [here](https://github.com/echasnovski/mini.nvim/blob/c8922aef8207137e66c80acdb9523668599ba62a/doc/mini-pairs.txt#L324-L328).
- Pressing `<BS>` inside registered pair (for example, `()`) leads to deleting whole pair. See [here](https://github.com/echasnovski/mini.nvim/blob/c8922aef8207137e66c80acdb9523668599ba62a/doc/mini-pairs.txt#L294-L298).

Hope this helps.

**Reply by jiggol** [reply]
Thank you very much.
